api/org/commcare/api/util/UserDataUtils.java

```diff
@@ -0,0 +1,178 @@
+package org.commcare.api.util;
+
+import org.commcare.api.interfaces.UserDataInterface;
+import org.commcare.api.persistence.SqlSandbox;
+import org.commcare.data.xml.DataModelPullParser;
+import org.commcare.suite.model.User;
+import org.commcare.api.parser.CommCareTransactionParserFactory;
+import org.javarosa.core.api.NameHasher;
+import org.javarosa.core.model.condition.EvaluationContext;
+import org.javarosa.core.model.instance.AbstractTreeElement;
+import org.javarosa.core.model.instance.DataInstance;
+import org.javarosa.core.model.instance.FormInstance;
+import org.javarosa.core.model.instance.TreeReference;
+import org.javarosa.core.services.storage.IStorageIterator;
+import org.javarosa.core.services.storage.IStorageUtilityIndexed;
+import org.javarosa.core.util.ArrayUtilities;
+import org.javarosa.core.util.externalizable.PrototypeFactory;
+import org.javarosa.model.xform.XPathReference;
+import org.javarosa.xml.util.InvalidStructureException;
+import org.javarosa.xml.util.UnfullfilledRequirementsException;
+import org.xmlpull.v1.XmlPullParserException;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Date;
+import java.util.Vector;
+
+/**
+ * Methods that mostly are used around the mocks that replicate stuff from
+ * other projects.
+ *
+ * TODO: We should try to centralize how these are used.
+ *
+ * @author ctsims
+ */
+public class UserDataUtils {
+
+    public static SqlSandbox getStaticStorage(String username) {
```

Comment:
I would move "usages of sandbox" and "retrieval of sandbox" into different categories